miden-lib/src/accounts/faucets/mod.rs

```diff
@@ -90,6 +90,7 @@ impl From<BasicFungibleFaucet> for AccountComponent {
 /// - Slot 2: Token metadata of the faucet.
 pub fn create_basic_fungible_faucet(
     init_seed: [u8; 32],
+    anchor_block_epoch_and_hash: (u16, Digest),
     symbol: TokenSymbol,
     decimals: u8,
     max_supply: Felt,
```

Comment:
Maybe a question for the client, @igamigo:

To create an account we now need 1) an anchor epoch and 2) the block hash of the anchor epoch block. (Basically the block hash of block $0 * 2^{16}$, $1 * 2^{16}$, $2 * 2^{16}$, ...). So the epoch and the hash are logically bound to each other. There are generally two ways to specify this in `AccountBuilder`. Either setting each value individually using `AccountBuilder::anchor_epoch` and `AccountBuilder::anchor_block_hash` or specifying both at once by passing a `BlockHeader` to `AccountBuilder::anchor_block_header`. The question is what is best to do here for these convenience functions?

Is it fine to require a `BlockHeader` or would that be too restrictive for some reason? My hunch is that the client or any other real world usage would likely always take these values from a block header anyway, so we might as well make it safer and require it. That would be my preferred way to go.
I'm not a huge fan of the current tuple approach, but it is the most flexible (for tests, etc.). I made it a tuple because of the logical connection between the two so making it two separate arguments is probably worse.

Reply:
Relatedly, I'm also wondering about `Account::new`. I just fixed it so it compiles, but it takes the anchor epoch and hash separately as well and is probably not very usable as it is. Interestingly, the method is unused in miden-base, miden-client and miden-node as far as I can tell. So I think we can just remove it and point to `AccountBuilder` for building accounts.

Reply:
Requiring a block header should be fine. Just to clarify, you mean specifically in functions such as `create_basic_fungible_faucet()`, correct? 
On `Account::new()`, I also agree on removing it.

Reply:
> Just to clarify, you mean specifically in functions such as `create_basic_fungible_faucet()`, correct?

Yes, this just relates to `create_basic_fungible_faucet` and `create_basic_wallet`.